tests/end_to_end/netsh_test.cpp

```diff
@@ -378,7 +378,7 @@ TEST_CASE("set program", "[netsh][programs]")
     REQUIRE(UuidToStringW(&EBPF_ATTACH_TYPE_XDP, &attach_type_string) == 0);
 
     // Attach the program.
-    output = _run_netsh_command(handle_ebpf_set_program, L"196609", (PCWSTR)attach_type_string, nullptr, &result);
+    output = _run_netsh_command(handle_ebpf_set_program, L"196609", L"xdp", nullptr, &result);
```

Comment:
Curious, what was the issue here?  The original code looks correct.

Reply:
Here is the parameter description for `netsh ebpf set program` 
```
\n      attached  - A string used as if it were a section name, to\
\n                  determine the attach type.  Use an empty string\
\n                  to detach it.\
```

But the test passed attach type GUID string instead of section name to `handle_ebpf_set_program`. So the test was always wrong.

`handle_ebpf_set_program` calls `ebpf_get_program_type_by_name  -> get_program_type_windows(const std::string& section, const std::string&)`
The reason why `get_program_type_windows` worked is because it uses the global program type `const ebpf_program_type_t* program_type = get_global_program_type();` if one is set. After that `ebpf_get_program_type_by_name ` got the attach type like this:
`*expected_attach_type = data->attach_type_uuid;`

But now that we can have multiple attach types per program type, the above was changed to get the attach type from a mapping between section name and attach type. So I had to fix the test case to pass the correct parameter - which would be section name.